### ISSUE TITLE
Can't reproduce results on fine-tuned checkpoint

### ISSUE THREAD

**Issue by oljike** [Lakonik/piFlow#16]
HI! Thank again for your work.

Currently I'm trying to reproduce your results on QWEN image. However I'm loading my fine-tuned checkpoint, not the original weights. 
The results I get are very bad. Also I noticed that the model achieves it's peak quality by step 200, after which it starts to degrade. 
Do you have any ideas what I should change to train fintetuned model.

ps. my finetuned checkpoint produce great images as it is.

**Comment by oljike** [comment]
The config I used:
_base_ = ['./_ddp_train.py', './_data_trainval.py']

name = 'gmqwen_baseline'

model = dict(
    type='LatentDiffusionTextImage',
    vae=dict(
        type='PretrainedVAEQwenImage',
        from_pretrained='Qwen/Qwen-Image',
        subfolder='vae',
        freeze=True,
        torch_dtype='bfloat16'),
    diffusion=dict(
        type='PiFlowImitationDataFree',
        policy_type='GMFlow',
        denoising=dict(
            type='GMQwenImageTransformer2DModel',
            patch_size=2,
            freeze=True,
            freeze_exclude=[
                'proj_out_means',
                'proj_out_logweights',
                'proj_out_logstds',
                'norm_out',
                'lora'],
            pretrained='huggingface://Qwen/Qwen-Image/transformer/diffusion_pytorch_model.safetensors.index.json',
            num_gaussians=8,
            logstd_inner_dim=1024,
            gm_num_logstd_layers=2,
            logweights_channels=4,
            in_channels=64,
            out_channels=64,
            num_layers=60,
            attention_head_dim=128,
            num_attention_heads=24,
            joint_attention_dim=3584,
            axes_dims_rope=(16, 56, 56),
            torch_dtype='bfloat16',
            checkpointing=True,
            use_lora=True,
            lora_target_modules=[
                'img_mlp.net.0.proj',
                'img_mlp.net.2',
                'timestep_embedder.linear_1',
                'timestep_embedder.linear_2'
            ] + [
                f'transformer_blocks.{i}.txt_mlp.net.0.proj' for i in range(59)
            ] + [
                f'transformer_blocks.{i}.txt_mlp.net.2' for i in range(59)
            ],
            lora_dropout=0.05,
            lora_rank=256),
        flow_loss=dict(
            type='DiffusionMSELoss',
            data_info=dict(pred='u_t_pred', target='u_t'),
            rescale_mode='constant',
            rescale_cfg=dict(scale=30.0)),
        num_timesteps=1,
        timestep_sampler=dict(
            type='ContinuousTimeStepSampler',
            shift=3.2,
            logit_normal_enable=False),
        denoising_mean_mode='U'),
    diffusion_use_ema=True,
    teacher=dict(
        type='GaussianFlow',
        denoising=dict(
            type='QwenImageTransformer2DModel',
            patch_size=2,
            freeze=True,
            pretrained='huggingface://Qwen/Qwen-Image/transformer/diffusion_pytorch_model.safetensors.index.json',
            in_channels=64,
            out_channels=64,
            num_layers=60,
            attention_head_dim=128,
            num_attention_heads=24,
            joint_attention_dim=3584,
            axes_dims_rope=(16, 56, 56),
            torch_dtype='bfloat16'),
        num_timesteps=1,
        denoising_mean_mode='U'),
    tie_teacher=True,
)

save_interval = 100
must_save_interval = 200  # interval to save regardless of max_keep_ckpts
eval_interval = 100
work_dir = f'work_dirs/{name}'
# yapf: disable
train_cfg = dict(
    num_decay_iters=2000,
    window_substeps=3,
    gm_dropout=0.1,
    num_intermediate_states=4,
    teacher_guidance_scale=4.0,
    # teacher_distilled_guidance_scale=4.0,
    # distilled_guidance_scale=4.0,
    nfe=4,
    final_step_size_scale=0.5,
    total_substeps=128,
)
test_cfg = dict(
    nfe=4,
    final_step_size_scale=0.5,
    total_substeps=128,
)

data = dict(
    workers_per_gpu=2,
    train_dataloader=dict(samples_per_gpu=2),
    val_dataloader=dict(samples_per_gpu=1),
    test_dataloader=dict(samples_per_gpu=1),
    persistent_workers=True,
    prefetch_factor=2
)
checkpoint_config = dict(
    interval=save_interval,
    must_save_interval=must_save_interval,
    by_epoch=False,
    max_keep_ckpts=1,
    out_dir='checkpoints/')

evaluation = []
for data_split in ['val']:
    for temperature in [0.3]:
        prefix = f'step4_temp{temperature}'
        evaluation.append(
            dict(
                type='GenerativeEvalHook',
                data=data_split,
                prefix=prefix,
                sample_kwargs=dict(
                    test_cfg_override=dict(
                        temperature=temperature,
                    )),
                interval=eval_interval,
                viz_dir=None,
                metric_cpu_offload=True,
                save_best_ckpt=False,
                metrics=dict(
                    type='IS',
                    num_images=256,  # Number of images to use for calculation (e.g., 5k or 50k)
                    inception_args=dict(
                        type='StyleGAN',
                        inception_path='/home/ubuntu/checkpoints/inception-2015-12-05.pt'
                    )
                )
                
                ))

total_iters = 9000
log_config = dict(
    interval=10,
    hooks=[
        dict(type='TextLoggerHook'),
        dict(type='TensorboardLoggerHook'),

        dict(
            type='WandbLoggerHook',
            init_kwargs=dict(
                project='attacka_na_karphagen',  # Name of your WandB project
                name=name,                 # Uses 'gmqwen_baseline' defined at the top
                entity='higgsfield-ai',  # Optional: User or Team name
                config=dict(               # Optional: Log extra hyperparams explicitly
                    work_dir=work_dir,
                    train_cfg=train_cfg,
                    data=data
                )
            )
        ),
    ])
# yapf:enable

custom_hooks = [
    dict(
        type='ExponentialMovingAverageHookMod',
        module_keys=('diffusion_ema', ),
        interp_mode='lerp',
        interval=1,
        start_iter=100,
        momentum_policy='karras',
        momentum_cfg=dict(gamma=7.0),
        priority='VERY_HIGH'),
]

load_from = None
resume_from = f'checkpoints/{name}/latest.txt'  # resume by default
workflow = [('train', save_interval)]



I have 8 B200 GPUs. 

However, it's worth mentioning that i conducted multiple experiments, with different lr to scale to my number of gpu. Also, used different accumulation steps. 

> Do you mean a finetuned base model instead of the official qwen model?
- Yes, i finetuned the official model weights, not changing anything in the architecture.

**Comment by Lakonik** [comment]
> Yes, i finetuned the official model weights, not changing anything in the architecture.

Thanks for the clarification. So you are still using the official qwen weights as the teacher.

A couple of things I noticed in your config

**num_decay_iters=2000**
Do NOT use this setting for the Qwen model. Scheduled trajectory mixing is only for the FLUX model. On other models it may make things worse.


**num_intermediate_states=4**
This is OK but likely unnecessary. It increases the overhead but likely will not make a big different to the results.

**metrics**
Do not use IS metric for non-ImageNet models. You may use HPSv2 to monitor convergence, e.g., 
```
           metrics=[dict(
                    type='HPSv2',
                    hps_version='v2.1')]
```
But still we recommend directly viewing the visualized images yourself---no metric is perfect!

Other things should be fine. Note that if you are using 8 GPUs, the effective batch size would be 8 * samples_per_gpu * nfe = 64, which is smaller than our 32 * samples_per_gpu * nfe = 256. We have tested batch size = 64 and visually the results look just fine. If you really want to reproduce the exact setting, you can set samples_per_gpu=8 to match the original batch size.

**Comment by oljike** [comment]
The problem is that after trained 9000 steps, the quality is the same. 

What can I change in the config to successfully train upon the finetuned checkpoint?

**Comment by oljike** [comment]
DX policy also did not work.

Do any parameters of config file come to your which I can change to train my finetuned checkpoint?
